typed-racket-lib/typed-racket/base-env/base-env.rkt

```diff
@@ -1951,9 +1951,23 @@
 ;; Section 13.1.10.2
 [input-port-append  (->* (list Univ) -Input-Port -Input-Port)]
 
-;TODO write the type for this
-;It is fairly complicated and require events
-;make-input-port/read-to-peek
+[make-input-port/read-to-peek
+ (->opt Univ
+        (-> -Bytes (Un -Nat (-val eof) top-func (-evt -Zero)))
```

Comment:
Sorry for the delay here. I think we should change the types to make it more clear, rather than using `top-func`.